libdrgn/dwarf_info_cache.c

```diff
@@ -88,6 +88,26 @@ static int dwarf_flag(Dwarf_Die *die, unsigned int name, bool *ret)
 	return dwarf_formflag(attr, ret);
 }
 
+/**
+ * Deduce language which corresponds with the CU of the given dwarf die.
+ *
+ * @param[in] die Dwarf_Die used to retrieve the current language.
+ * @param[out] lang Language to return
+ * @return @c NULL on success, non-@c NULL on error.
+ */
+static struct drgn_error *drgn_language_from_die(Dwarf_Die *die, const struct drgn_language **lang)
+{
+	Dwarf_Die cudie;
+	if (dwarf_cu_info(die->cu, NULL, NULL, &cudie, NULL, NULL, NULL, NULL) == 0) {
+		// TODO don't fail silently on failure
+		*lang = drgn_language_from_dw(dwarf_srclang(&cudie));
+		return NULL;
+	}
+	*lang = &drgn_default_language;
+	return drgn_error_format(DRGN_ERROR_LOOKUP,
+				 "Binary is missing CU information.");
```

Comment:
This is a fatal error, so we don't need to return anything in `*lang`. Also, we should return the error from libdw with `drgn_error_libdw()`.